core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -348,6 +350,11 @@ public void syncPostBatch(
             int shortCircuited = 0;
             final Map<BatchItem, CompletableFuture<Exception>> sendFutures = new HashMap<>();
             for (final BatchItem item : batch) {
+                if (item.getBrokerId() == null) {
+                    item.updateStatusAndDetail(EventPublishingStatus.FAILED, String.format("No leader for partition: %s, topic: %s.", item.getPartition(), topicId));
+                    LOG.warn("Failed to publish. No leader fo partition [{}], topic [{topicId}].", item.getPartition(), topicId);
```

Comment:
```suggestion
                    LOG.warn("Failed to publish to Kafka. No leader for partition [{}], topic [{topicId}].", item.getPartition(), topicId);
```

Reply:
and probably it will be a lot of messages, maybe we want to cut it somehow

Reply:
maybe the same way as for short circuiting

Reply:
https://github.com/zalando/nakadi/blob/master/core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java#L367 This doesn't look too short.

This event usually doesn't happen, but if it does wouldn't it be better to have more logs?

Reply:
I mean log less lines. compact them in some way, because most probably there will be a lot of same lines which we can avoid logging and the same lines do not bring any value.

Reply:
I prefer to see logs as long as the issue is happening. You can aggregate log lines to see what is the severity and once the logline stops appearing you know that the problem is gone.

Reply:
my suggestion does not exclude logging while issues is happening. you will still be able to see logs, just logs you need and not repeating once. as we know the issue can persist for hours eating resources while logging.